### PR TITLE
[FIX] l10n_*: set default taxes

### PR DESCRIPTION
**Pull request by dbkosky**
The default taxes for most localisations have been left undefined by default. When loading the chart template, the model generally selects the first sales and purchase taxes, based on the order in which the taxes appear in the csv, for the default sales and purchase taxes respectively.

This behaviour can be confusing to those who are not yet familiar with it. It has been decided that it is preferable instead to specify the default tax in _get_*_res_company function on the account chart template model, such that the 
default taxes are defined explicitly for every localisation.

linked enterprise pr https://github.com/odoo/enterprise/pull/45531

task-3453997

### CODE REVIEW

**Comment by dbkosky** [comment]
@JulienAlardot Sorry :wink: 
(if everything looks good to you, we can tag oco, since he is aware of this one :slightly_smiling_face: )

File: addons/l10n_ca/models/template_ca.py [hunk 1]
```diff
@@ -21,26 +21,19 @@ def _get_ca_template_data(self):
...
+            'ON': ('hst_sale_tax_13', 'hst_purchase_tax_13'),
+            'NB': ('hst_sale_tax_15', 'hst_purchase_tax_15'),
+            'NL': ('hst_sale_tax_15', 'hst_purchase_tax_15'),
+            'NS': ('hst_sale_tax_15', 'hst_purchase_tax_15'),
+            'PE': ('hst_sale_tax_15', 'hst_purchase_tax_15'),
+        }.get(self.env.company.state_id.code, ('gst_sale_tax_5', 'gst_purchase_tax_5'))
 
-        default_sales_tax, default_purchase_tax = get_default_taxes()
         return {
             company.id: {
```
Comment (JulienAlardot) [inline addons/l10n_ca/models/template_ca.py:38]:
```suggestion
            self.env.company.id: {
```
oops, my bad for that one xD

**Review by JulienAlardot — CHANGES_REQUESTED** [review]

File: addons/l10n_ca/models/template_ca.py (hunk 1 above)
Comment (dbkosky) [reply addons/l10n_ca/models/template_ca.py]: ah thanks :)

**Comment by JulienAlardot** [comment]
LGTM @jbw-odoo

**Comment by dbkosky** [comment]
@JulienAlardot I'm tagging @oco-odoo on this one, since we discussed it on Tuesday

**Comment by JulienAlardot** [comment]
> @JulienAlardot I'm tagging @oco-odoo on this one, since we discussed it on Tuesday

Yeah, right
mentionning JB is a reflex now xD

**Comment by Levizar** [comment]
l10n standalone is red, probably a test issue

File: addons/l10n_in_edi/tests/test_edi_json.py [hunk 2]
```diff
@@ -31,16 +31,22 @@ def setUpClass(cls, chart_template_ref="in"):
...
             'name': 'product_a2',
             'uom_id': cls.env.ref('uom.product_uom_unit').id,
             'lst_price': 1000.0,
             'standard_price': 1000.0,
             'property_account_income_id': cls.company_data['default_account_revenue'].id,
             'property_account_expense_id': cls.company_data['default_account_expense'].id,
-            'taxes_id': [(6, 0, cls.tax_sale_a.ids)],
-            'supplier_taxes_id': [(6, 0, cls.tax_purchase_a.ids)],
+            'taxes_id': [(6, 0, sgst_sale_5.ids)],
+            'supplier_taxes_id': [(6, 0, sgst_purchase_5.ids)],
```
Comment (Levizar) [inline addons/l10n_in_edi/tests/test_edi_json.py:49]:
```suggestion
            'taxes_id': [Command.set(sgst_sale_5.ids)],
            'supplier_taxes_id': [Command.set(sgst_purchase_5.ids)],
```

File: addons/l10n_in_edi/tests/test_edi_json.py [hunk 3]
```diff
@@ -50,8 +56,8 @@ def setUpClass(cls, chart_template_ref="in"):
             'standard_price': 400.0,
             'property_account_income_id': cls.company_data['default_account_revenue'].id,
             'property_account_expense_id': cls.company_data['default_account_expense'].id,
-            'taxes_id': [(6, 0, cls.tax_sale_a.ids)],
-            'supplier_taxes_id': [(6, 0, cls.tax_purchase_a.ids)],
+            'taxes_id': [(6, 0, sgst_sale_5.ids)],
+            'supplier_taxes_id': [(6, 0, sgst_purchase_5.ids)],
```
Comment (Levizar) [inline addons/l10n_in_edi/tests/test_edi_json.py:60]: `Command.Set`

File: addons/l10n_in_edi/tests/test_edi_json.py [hunk 4]
```diff
@@ -64,7 +70,7 @@ def setUpClass(cls, chart_template_ref="in"):
             "property_account_income_id": cls.company_data["default_account_revenue"].id,
             "property_account_expense_id": cls.company_data["default_account_expense"].id,
             "taxes_id": [(6, 0, gst_with_cess.ids)],
-            "supplier_taxes_id": [(6, 0, cls.tax_purchase_a.ids)],
+            "supplier_taxes_id": [(6, 0, sgst_purchase_5.ids)],
```
Comment (Levizar) [inline addons/l10n_in_edi/tests/test_edi_json.py:73]: same

**Review by Levizar — CHANGES_REQUESTED** [review]
GJ!

just a few unimportant comments.
Also, there is a conflict with l10n_ZA to fix

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**